### PR TITLE
GH-39664: [C++][Acero] Ensure Acero benchmarks present a metric for identifying throughput

### PR DESCRIPTION
**Pull request by llama90**
### Rationale for this change

Acero benchmarks sometimes output metrics such as `items/s`, `bytes/s`, `batches/s`, and `rows/s`. However, there is inconsistency in how these metrics are presented across different benchmarks. We are undertaking an effort to standardize the output of these metrics to ensure uniformity and clarity in performance measurement across all Acero benchmarks. 

### What changes are included in this PR?

`rows/s` has a similar meaning to `items/s`.

- `bytes/s` and `items/s`: aggregate
- `bytes/s` and `rows/s`: asof_join
- `batches/s` and `rows/s`: project, filter, expression

### Are these changes tested?

Yes.

### Are there any user-facing changes?

No.

* GitHub Issue: #39664

### CODE REVIEW

**Comment by pitrou** [comment]
Note that `bytes_per_second` and `items_per_second` are standard metrics with Google benchmark, so they will have to stay. For Acero, I think we should go with the convention that 1 "item" == 1 row. As for `bytes_per_second`, it's less useful than `items_per_second` IMHO, but can be kept if it already exists.

**Review by westonpace — CHANGES_REQUESTED** [review]
A few minor concerns

File: cpp/src/arrow/acero/aggregate_benchmark.cc [hunk 1]
```diff
@@ -321,6 +321,19 @@ BENCHMARK_TEMPLATE(ReferenceSum, SumBitmapVectorizeUnroll<int64_t>)
 // GroupBy
 //
 
+int64_t CalculateBatchSize(const std::shared_ptr<RecordBatch>& batch) {
```
[the remark this answers is not in the document]
Comment (llama90) [reply cpp/src/arrow/acero/aggregate_benchmark.cc]: I fixed it to replace CalculateBatchSize with TotalBufferSize.

Thank you for letting me know.

File: cpp/src/arrow/acero/expression_benchmark.cc [hunk 2]
```diff
@@ -66,17 +69,34 @@ static void BindAndEvaluate(benchmark::State& state, Expression expr) {
...
   for (auto _ : state) {
     ABORT_NOT_OK(SimplifyWithGuarantee(filter, guarantee));
   }
+
+  state.counters["rows_per_second"] = benchmark::Counter(
+      static_cast<double>(state.iterations() * num_batches * rows_per_batch),
+      benchmark::Counter::kIsRate);
+
+  state.counters["batches_per_second"] = benchmark::Counter(
+      static_cast<double>(state.iterations() * num_batches), benchmark::Counter::kIsRate);
```
[the remark this answers is not in the document]
Comment (llama90) [reply cpp/src/arrow/acero/expression_benchmark.cc]: I rolled back all the changes related to `expressions_benchmark.cc`. As you mentioned, the metrics such as `rows_per_second` and `batches_per_second` seem unnecessary for this benchmark. While working, I was too focused on outputting these metrics.

Based on your review, it appears that those were incorrect modifications, so I have rolled back all related changes.

Thank you for the review.

File: cpp/src/arrow/acero/asof_join_benchmark.cc [hunk 3]
```diff
@@ -91,7 +91,7 @@ static void TableJoinOverhead(benchmark::State& state,
     ASSERT_OK(DeclarationToStatus(std::move(join_node), /*use_threads=*/false));
   }
 
-  state.counters["input_rows_per_second"] = benchmark::Counter(
+  state.counters["rows_per_second"] = benchmark::Counter(
```
Comment (llama90) [inline cpp/src/arrow/acero/asof_join_benchmark.cc:94]: All the other codes also use the `rows_per_second` value, so I made the modifications accordingly.

**Review by westonpace — APPROVED** [review]

**Pull request merged**